task2.py

```diff
@@ -0,0 +1,71 @@
+# Person names
+names = [
+    "Smith",
+    "Johnson",
+    "Williams",
+    "Jones",
+    "Brown",
+    "Davis",
+    "Miller",
+    "Wilson",
+    "Moore",
+    "Taylor",
+    "Anderson",
+    "Thomas",
+    "Jackson",
+    "White",
+]
+#Create dictionary name: name length
+names_length = {name: len(name) for name in names}
+print("Length of names", list(names_length.values()))
+
+# Sorting 
+sorted_names_length = sorted(names_length.items(), key=lambda x: x[1])
+
+
+#Create dictionary name length vise value
+length_wise_names = dict()
+for name_length_tuple in sorted_names_length:
+    if name_length_tuple[1] in length_wise_names.keys():
+        length_wise_names[name_length_tuple[1]].append(name_length_tuple[0])
+    else:
+        length_wise_names[name_length_tuple[1]] = [name_length_tuple[0]]
+
+
+# 3 most frequent names
+print("\n The three most frequent name lenghts are:")
+three_most_freq_names = sorted(length_wise_names, key=length_wise_names.get, reverse=True)[:3]
+for top_three in three_most_freq_names:
+    if top_three == 5:
+        print("6 names of length 5 :-",length_wise_names[top_three])
+    elif top_three == 6:
+        print("4 names of length 6 :-",length_wise_names[top_three])
+    elif top_three == 8:
+        print("2 names of length 8 :-",length_wise_names[top_three])
+
+# 3 least frequent names
+print("\n The three least frequent name lenghts are:")
+three_least_freq_names = sorted(length_wise_names, key=length_wise_names.get, reverse=True)[-3:]
+for least_three in three_least_freq_names:
+    if least_three == 5:
+        print("6 names of length 5 :-",length_wise_names[least_three])
+    elif least_three == 6:
+        print("4 names of length 6 :-",length_wise_names[least_three])
+    elif least_three == 7:
+        print("2 names of length 7 :-",length_wise_names[least_three])
+
```

Comment:
Change the logic. This is wrong logic. What happends if most three frequent names are 100, 101, and 102. So please make it dynamic, this is still static.